### PR TITLE
Core, Flink: Fix build warnings

### PR DESCRIPTION
**Pull request by nk1506**

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/view/BaseViewOperations.java [hunk 1]
```diff
@@ -104,7 +104,7 @@ public ViewMetadata refresh() {
   @Override
   public void commit(ViewMetadata base, ViewMetadata metadata) {
     // if the metadata is already out of date, reject it
-    if (base != current()) {
+    if (!current().equals(base)) {
```
Comment (nastra) [inline core/src/main/java/org/apache/iceberg/view/BaseViewOperations.java:107]: the previous check is actually correct

File: core/src/main/java/org/apache/iceberg/view/BaseViewOperations.java [hunk 2]
```diff
@@ -115,7 +115,7 @@ public void commit(ViewMetadata base, ViewMetadata metadata) {
     }
 
     // if the metadata is not changed, return early
-    if (base == metadata) {
+    if (metadata.equals(base)) {
```
Comment (nastra) [inline core/src/main/java/org/apache/iceberg/view/BaseViewOperations.java:118]: the previous check is actually correct

File: core/src/main/java/org/apache/iceberg/view/BaseViewOperations.java (hunk 1 above)
Comment (nk1506) [reply core/src/main/java/org/apache/iceberg/view/BaseViewOperations.java:107]: Screenshot 2024-08-08 at 6 49 29 PM

I am seeing above warning with build. Is it ignorable ?

Comment (nastra) [reply core/src/main/java/org/apache/iceberg/view/BaseViewOperations.java:107]: yes, in this case this warning should be ignored

**Review by amogh-jahagirdar — APPROVED** [review]
Thanks @nk1506

File: core/src/main/java/org/apache/iceberg/view/BaseViewOperations.java (hunk 1 above)
Comment (nastra) [reply core/src/main/java/org/apache/iceberg/view/BaseViewOperations.java]: @nk1506 are you going to add a suppresswarning for this case?

File: core/src/main/java/org/apache/iceberg/view/BaseViewOperations.java (hunk 2 above)
Comment (nastra) [reply core/src/main/java/org/apache/iceberg/view/BaseViewOperations.java]: @nk1506 you'll need to add a suppresswarning in this case to the method

**Review by nastra — APPROVED** [review]

**Pull request merged**